src/main/java/com/iexec/core/result/ResultService.java

```diff
@@ -55,18 +54,17 @@ public boolean isResultUploaded(String chainTaskId) {
             return false;
         }
 
-        return resultRepoClient.isResultUploaded(resultProxyToken, chainTaskId)
-                .getStatusCode()
-                .is2xxSuccessful();
+        resultProxyClient.isResultUploaded(resultProxyToken, chainTaskId);
```

Comment:
It is not about `isResultUploaded` being `false`, it is about its http status being 2xx or not.

On iexec-result-proxy side, the important part is here:
https://github.com/iExecBlockchainComputing/iexec-result-proxy/blob/develop/src/main/java/com/iexec/resultproxy/proxy/ProxyController.java#L117
We can only have a 2xx http status if result has been uploaded (http 204 no content).
It will be 404 not found or 401 unauthorized otherwise.

With the new feign clients, non-2xx http status will throw `FeignException` which will be managed with `spring-retry` annotations in iexec-core (`@Retryable` and `@Recover`).

Here, with spring-retry, when everything is ok, no exception happens and we can return true.
When errors happen several times in a row, we go to the recovery method and return a global false stating that upload failed several times and cannot be performed.
In both cases, we ignore the empty body and only return a boolean.

Note : No body is provided in iexec-result-proxy `ProxyController#isResultUploaded`.
The method signatures may be misleading.

Reply:
I just noticed `public ResponseEntity<**String**> isResultUploaded(..)` yes, this isn't meaningful to me yes